src/og/control/LayerSwitcherv2.js

```diff
@@ -0,0 +1,490 @@
+            if (whereIclicked == 'outside' && D_header_pin.dataset.attachement == 'UNPINNED') {
+                D_dialog.classList.add('og-not-visible')
+                D_menu_btn.classList.add('og-OFF')
+                // If Iclicked on button, toggle dialog and menu btn
+            } else if (whereIclicked == 'on-btn') {
+                D_dialog.classList.toggle('og-not-visible')
+                D_menu_btn.classList.toggle('og-OFF')
+            }
+        }
+
+        // Document
+        document.addEventListener('click', (e) => whereClickHandler(e))
+
+        // Header pin - here I am using instead of CSS classes, the data-* attribute of the DOM element to store the state
+        D_header_pin.addEventListener('click', () => {
+            let newText = toggleText(D_header_pin.dataset.attachement, ['PINNED', 'UNPINNED'])
+            D_header_pin.classList.toggle('og-OFF')
+            D_header_pin.dataset.attachement = newText
+        })
+
+        // Header double click
+        D_header.addEventListener('dblclick', (e) => {
+            if (D_header !== e.target) return
+            restoreInitialPos(D_dialog)
+        }, false)
+
+        const restoreInitialPos = (el) => {
+            el.style.top = dialogBoxInitial.top + "px";
+            el.style.left = dialogBoxInitial.left + "px";
+        }
+
+        // Header mousedown --> move dialog OR nothing
+        const { behaviour, mouseMove } = enableElmovement(D_dialog, this.planet)
+
+        D_header.addEventListener('mousedown', (e) => {
+            if (D_header_pin.dataset.attachement == 'UNPINNED') {
+                behaviour(e)
+            } else {
+                document.removeEventListener('mousemove', mouseMove)
+            }
+        })
+
+        D_header_close.addEventListener('click', () => {
+            D_dialog.classList.add('og-not-visible')
+            D_menu_btn.classList.add('og-OFF')
+        })
+
+        D_header_minMax.addEventListener('click', () => {
+            D_main_container.classList.toggle('og-hide')
+            D_main_container.classList.toggle('og-minimized')
+            D_dialog.classList.toggle('og-minimized')
+        })
+
+        return D_main_container
+    }
+
+    buildRecords(myData, mainContainer, depth, createLastDropZone) {
+        let planet = this.planet
+
+        // Record functions
+        const createTitle = (object) => {
+            if (object.data) {
+                return elementFactory('details', { class: 'og-layer-switcher-record-title' },
+                    elementFactory('summary', {}, object.name || object.url || 'no-name'))
+            } else {
+                return elementFactory('label', { class: 'og-layer-switcher-record-title' },
+                    object.name || object.properties.name || 'no-name')
+            }
+        }
+
+        const visibility = (object, nameConcat) => {
+            if (nameConcat == 'TerrainProviders' && planet.terrain == object) {
```

Comment:
maybe it is better to refactor these cases into one?

Reply:
Yes, noticed it also my self